src/api/c++/z3++.h

```diff
@@ -567,8 +623,7 @@ namespace z3 {
             ast_vector_tpl const* m_vector;
             unsigned m_index;
         public:
-            iterator(ast_vector_tpl const* v, unsigned i): m_vector(v), m_index(i) {}
-            iterator(iterator const& other): m_vector(other.m_vector), m_index(other.m_index) {}
+            iterator(ast_vector_tpl const* v, unsigned i) noexcept : m_vector(v), m_index(i) {}
```

Comment:
@NikolajBjorner I left this constructor in, but constructors like this are not needed. C++11 introduces initializer lists which have a few benefits. For example, `A a{1, 2}` would correctly initialize `struct A { int a; int b };` The compiler will implicitly generate a constructor with `noexcept` and everything it can. Having explicit constructors for pure data copying like this is actually against cpp guidelines. That said if we are upping to C++11 just now, people using the older `()` syntax for constructors might run into a harder time; that's why I left it in. If you have any objections to this though since it is bad practice let me know.

Reply:
It seems like nit. I say leave it at what it is now and move on to more consequential issues